### ISSUE TITLE
Map class names with their probabilities

### ISSUE THREAD

**Issue by shamindraparui** [ultralytics/ultralytics#4962]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,
I have trained Yolo-v8 image classification with 2 classes. During inference, it is returning probabilities as a list. Code is:
`for each in results:`
 `   print(each.probs.numpy().data)`

and result is:
`[   0.018698      0.9813]`
How can I map those probabilities to their class names?

**Comment by RizwanMunawar** [comment]
Hello @shamindraparui, the provided code can be utilized for associating the probabilities with their corresponding class names. This allows you to establish a clear mapping between the model's output and the classes it predicts.

```
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n-cls.pt')
model.to("cpu")

# Predict with the model
results = model('https://ultralytics.com/images/bus.jpg')

classes_names = ["class_1", "class_2", "class_3"]

for each in results:
    probabilities = each.probs.numpy().data
    class_index = probabilities.argmax()  # Get the index with the highest prob
    class_name = classes_names[class_index]  # Map the index to the class name
    print(f"Class: {class_name}, Probability: {probabilities[class_index]}")
```

**Comment by shamindraparui** [comment]
@RizwanMunawar  should `classes_names` follow alphabetical order of the class names?

**Comment by RizwanMunawar** [comment]
@shamindraparui, the class names must match those specified in the configuration file, and they should adhere to the same order as the indexing.

**Comment by shamindraparui** [comment]
@RizwanMunawar the task is to classify the image, I am not performing object detection, so I have no configuration file. According to my understanding, Yolo V8 infer classes, during training process, by observing directory structure.

**Comment by RizwanMunawar** [comment]
@shamindraparui, the classification method also relies on a configuration file. Specifically, all 'yolov8-cls' models have been trained using the [ImageNet configuration file](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/ImageNet.yaml).

**Comment by shamindraparui** [comment]
@RizwanMunawar  I got your point, network needs the config which must contain the essentials. I went through the following 2 docs regarding classification:
[Image Classification](https://docs.ultralytics.com/tasks/classify/)
[Image Classification Datasets Overview](https://docs.ultralytics.com/datasets/classify/)

My training code is:
```
model = YOLO('yolov8m-cls.pt')
results = model.train(data='side_cam', epochs=5, imgsz=64)
```
Output produces by the last line is:

> train: ****\side_cam\train... found 130 images in 2 classes  
> val: None
> test: ****\side_cam\test... found 20 images in 2 classes  
> Overriding model.yaml nc=1000 with nc=2
>                    from  n    params  module              arguments                     
>   0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]                 
>   1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]                
>   2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]             
>   3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]               
>   4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]           
>   5                  -1  1    664320  ultralytics.nn.modules.conv.Conv             [192, 384, 3, 2]              
>   6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]           
>   7                  -1  1   2655744  ultralytics.nn.modules.conv.Conv             [384, 768, 3, 2]              
>   8                  -1  2   7084032  ultralytics.nn.modules.block.C2f             [768, 768, 2, True]           
>   9                  -1  1    988162  ultralytics.nn.modules.head.Classify         [768, 2]                      
> YOLOv8m-cls summary: 141 layers, 15774898 parameters, 15774898 gradients
> Transferred 228/230 items from pretrained weights
> optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 38 weight(decay=0.0), 39 weight(decay=0.0005), 39 bias(decay=0.0)
> Image sizes 64 train, 64 val
> Using 0 dataloader workers
> Logging results to runs\classify\train8

As the output says _Overriding model.yaml nc=1000 with nc=2_, which means it is replacing the default. So, how can I get the order of class names according to which it is being trained on?

**Comment by RizwanMunawar** [comment]
@shamindraparui Certainly, `model.names` will accurately supply the class names. This indicates that the network underwent training using the same methodology, and you can employ these names for subsequent processing.

**Comment by RizwanMunawar** [comment]
I am closing this issue. Please reopen if the issue persists

**Comment by shamindraparui** [comment]
@RizwanMunawar  yes, thanks.

**Comment by glenn-jocher** [comment]
@shamindraparui you're welcome! I'm glad I could help. If you have any other questions or if there's anything else you'd like to discuss, feel free to ask. Happy coding!
